### DISCUSSION TITLE
How can I determine a hidden window's size?

### DISCUSSION THREAD

**Discussion by Enyium** [slint-ui/slint#6608] [Q&A] [answered]
I need to center a newly displayed dialog window (from a `Dialog` .slint component) on another window (the app's main window). But the size of the dialog window before calling `show()`, as reported by `slint::Window::size() -> PhysicalSize`, is 800x600, which is incorrect and simply seems to be a default size on Windows.

Calling `Window::request_redraw()` before doesn't work. Calling `show()` directly before centering also doesn't work. As a workaround, I have to defer the centering on the main window with `dialog.as_weak().upgrade_in_event_loop(...)`, which means `show()` will run before.

Is a function that ensures layouting at a point in time required by the developer missing?

**Comment by hunger** [comment, marked as answer]
We do not seem to have a good solution for this: I took the liberty to open https://github.com/slint-ui/slint/issues/6724 for this.

Sorry for leaving this open for so long! That should not happen.

**Reply by Enyium** [reply]
Thank you!

There's another question from me still open: #5758. It's not urgent, but important for me. It would be great if this could be addressed at some point in the future, even if only one subtopic at a time. Perhaps, concrete issues for what's not quite supported yet could come out of this.
